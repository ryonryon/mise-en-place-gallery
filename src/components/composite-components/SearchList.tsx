import * as React from "react";

import UnderlineTextInput from "../inputs/UnderlineTextInput";
import BasicList from "../lists/BasicList";
import getRandomName from "../../services/getRandomName";
import InputType from "../inputs/input-type";

interface Props {
  isCaseIntensive: boolean;
}

const SearchList = ({ isCaseIntensive }: Props) => {
  const [fullNameList, setFullNameList] = React.useState([]);
  const [showingNameList, setShowingNameList] = React.useState([]);

  React.useEffect(() => {
    (async () => {
      const list = await getRandomName();
      setFullNameList(list);
      setShowingNameList(list);
    })();
  }, []);

  const handleChange = (event: React.FormEvent<HTMLInputElement>) => {
    const searchWord = event.currentTarget.value;
    setShowingNameList(
      fullNameList.filter(item =>
        isCaseIntensive ? item.toLowerCase().includes(searchWord.toLowerCase()) : item.includes(searchWord)
      )
    );
  };

  return (
    <>
      <UnderlineTextInput type={InputType.SEARCH} placeholder="search" onChange={handleChange} />
      <BasicList list={showingNameList} />
    </>
  );
};

SearchList.defaultProps = {
  isCaseIntensive: true
};

export default SearchList;
