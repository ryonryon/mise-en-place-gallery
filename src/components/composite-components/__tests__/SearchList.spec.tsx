import * as React from "react";
import * as renderer from "react-test-renderer";

import SearchList from "../SearchList";

describe("SearchList test", () => {
  it("tests SearchList renders correctly ", () => {
    const searchList = renderer.create(<SearchList />).toJSON();
    expect(searchList).toMatchSnapshot();
  });
});
