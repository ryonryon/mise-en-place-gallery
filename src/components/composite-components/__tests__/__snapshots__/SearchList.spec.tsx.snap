// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchList test tests SearchList renders correctly  1`] = `
Array [
  <input
    className="css-1ym2sj7"
    onChange={[Function]}
    placeholder="search"
    type="text"
  />,
  <div
    className="css-cm3tjs"
  >
    <ul />
  </div>,
]
`;
